cognite/client/utils/_identifier.py

```diff
@@ -107,7 +107,7 @@ def of_either(
                 instance_id = InstanceId.load(instance_id)
             if not isinstance(instance_id, InstanceId):
                 raise TypeError(f"Invalid instance_id, expected InstanceId, got {type(instance_id)}")
-        return Identifier(id or external_id or instance_id)
+        return Identifier(next(idx for idx in (id_, external_id, instance_id) if idx is not None))
```

Comment:
Thanks for fixing! Please add a test and add a comment like (example from `test_numpy_dtypes_conversions_for_string_and_numeric`) :
```
# Bug prior to 7.32.4, several methods on DatapointsArray would fail due to a bad
# conversion of numpy dtypes to native.
```